lib/matplotlib/axes/_base.py

```diff
@@ -3268,11 +3242,9 @@ def set_xticklabels(self, labels, fontdict=None, minor=False, **kwargs):
         """
         Set the x-tick labels with list of string labels.
 
-        .. ACCEPTS: list of string labels
-
         Parameters
         ----------
-        labels : list of str
+        labels : List[str]
```

Comment:
Did we somewhere agree to switch to type annotation syntax?

Reply:
Grepping for 'List\[' shows that it's definitely already being used.
```
lib/matplotlib/contour.py
576:        lw = self.get_label_width(self.labelLevelList[lmin],
577:                                  self.labelFmt, self.labelFontSizeList[lmin])
591:        self.add_label(xmin, ymin, rotation, self.labelLevelList[lmin],
592:                       self.labelCValueList[lmin])
1039:        artists : List[`.Artist`]
1042:        labels : List[str]
1810:        hatches : List[str], optional

lib/matplotlib/collections.py
386:        urls : List[str] or None
387:            .. ACCEPTS: List[str] or None

lib/matplotlib/tight_layout.py
52:    num1num2_list : List[int]

lib/matplotlib/widgets.py
515:        labels : List[str]
518:        actives : List[bool], optional
2484:    button : List[Int], optional

lib/matplotlib/backend_bases.py
1257:    callbacks : List[Tuple[callable, Tuple, Dict]]
2368:        callbacks : List[Tuple[callable, Tuple, Dict]]

lib/matplotlib/axes/_base.py
1613:        v : List[float] or one of the strings listed below.
```

Reply:
I'll accept this for now, even tough I would like to have it discussed and agreed upon. I'm not sure if this is really the best way. Whatever the policy is, it needs documentation. I'll leave the topic for #10225.